[docs] Update SO validate troubleshooting section for new PR comment format (#270927)
elastic/kibana: docs/extend/saved-objects/validate.md

> ### CI is failing for my PR

Reviewer: If you would like the rule type sections to be more visible in the TOC, you could:

- Have 2 level 2 (##) troubleshooting sections: 
   - Troubleshooting: CI failing on a PR
   - Troubleshooting: Kibana failing to start
- Promote the troubleshooting section here into its own page

With any of these changes you can bump most sections one level up so they show in the right nav

When the *Check changes in saved objects* CI step fails, a comment is posted directly on the PR. Each violation is grouped by SO type and shows a **rule ID**, a message, a fix hint, and a link to the relevant docs section:

> ```

Reviewer: ````suggestion
```markdown
````
You can get some highlighting for markdown too, if you think it can be valuable

### `<soType>`
- **[rule-id]** Message. _Fix:_ Fix hint. ([docs](link))
```

You can reproduce all findings locally:

```shell
# Get the merge-base with main (or the base branch of your PR)
git merge-base HEAD main

> # Run the check — add --fix to auto-generate missing fixture templates

Reviewer: ```suggestion
# Run the check. Add "--fix" to auto-generate missing fixture templates
```
Too many dashes got me confused there

node scripts/check_saved_objects --baseline <mergeBase> --fix
```

Use the rule IDs below to identify the problem and the appropriate fix.

> ---

Reviewer: Not a super big deal so feel free to keep but I don't know if those lines are super valuable in between each rule; it's at least not a pattern we use commonly in our docs.

I'd remove or at least maybe only have them between rule type sections instead of each rule

#### Existing type rules

##### `existing-type/mutated-migrations` [existing-type-mutated-migrations]

> **Problem:** The deprecated `migrations` property has been modified. It must remain unchanged so that old documents can still be imported.

Reviewer: 👍 I like the problem/scenario/fix layout you use

**Fix:** Revert any changes to `<type>.migrations`. If you need to change migration behavior, add a new model version instead.

---

##### `existing-type/mutated-existing-model-version` [existing-type-mutated-model-version]

**Problem:** A structural change was made to an already-defined model version (mappings, changes block, or similar). Existing model versions are immutable once merged.

> **Scenario 1:** You are fixing a bug in an existing model version. Once it is released (e.g. in Serverless), other nodes may already be running against it. Add a new model version instead of editing the existing one.

Reviewer: ```suggestion
**Scenario 1:** You are fixing a bug in an existing model version. Once it is released (for example, in Serverless), other nodes may already be running against it. Add a new model version instead of editing the existing one.
```

**Scenario 2:** You only added one new version but CI still reports a mutation. Validation uses two baselines: the PR merge-base and the **current Serverless release**. If the Serverless release includes a version that differs from your local copy, the check sees a mutation. Rebase onto the latest `main` to pick up any updates.

**Fix:** Revert the structural change and add a new model version to capture the update.

---

##### `existing-type/schema-breaking-changes` [existing-type-schema-breaking-changes]

> **Problem:** A breaking schema change was detected in an existing model version — for example, a field was removed, its type changed, or an optional field was made required. These changes would break documents already stored against that version.

Reviewer: ```suggestion
**Problem:** A breaking schema change was detected in an existing model version. For example, a field was removed, its type changed, or an optional field was made required. These changes would break documents already stored against that version.
```

---

#### New type rules

##### `new-type/missing-initial-model-version` [new-type-missing-initial-model-version]

**Problem:** A brand-new SO type does not define model version `1`.

**Fix:** Add a `modelVersions` entry with key `1` containing `schemas.create` and `schemas.forwardCompatibility`.

---

##### `new-type/legacy-migrations` [new-type-legacy-migrations]

**Problem:** A new SO type is using the deprecated `migrations` property.

**Fix:** Remove `migrations` and replace it with `modelVersions`, starting at version `1`.

---

##### `new-type/keyword-missing-ignore-above` [new-type-keyword-ignore-above]

Same cause and fix as [`existing-type/keyword-missing-ignore-above`](#existing-type-keyword-ignore-above), but for a brand-new type.

---

##### `new-type/invalid-name-title-field-type` [new-type-invalid-name-title]

Same cause and fix as [`existing-type/invalid-name-title-field-type`](#existing-type-invalid-name-title), but for a brand-new type.

---

#### Model version rules

These rules apply to both new and existing types.

##### `model-version/initial-must-be-schema-only` [model-version-initial-schema-only]

> **Problem:** The initial model version (`1`) of a new type defines a `changes` block (e.g. `mappings_addition`). For backward-compatibility reasons, the first model version can only contain `schemas`.

Reviewer: ```suggestion
**Problem:** The initial model version (`1`) of a new type defines a `changes` block (for example, `mappings_addition`). For backward-compatibility reasons, the first model version can only contain `schemas`.
```
